spec/services/service_offering/add_to_portfolio_item_spec.rb

```diff
@@ -1,104 +1,121 @@
-  let(:api_instance) { double }
   let(:service_offering_ref) { "1" }
-  let(:user_defined_name) { "Frank" }
-  let(:user_defined_description) { "Franks Description" }
-  let(:topo_ex) { Catalog::TopologyError.new("kaboom") }
   let(:subject) { described_class.new(params) }
-
-  let(:params) { HashWithIndifferentAccess.new(:name => user_defined_name, :description => user_defined_description, :service_offering_ref => service_offering_ref) }
-
-  let(:topology_service_offering) { fully_populated_service_offering }
-  let(:service_offering_icon) { fully_populated_service_offering_icon }
-  let(:topological_inventory) do
-    class_double("TopologicalInventory")
-      .as_stubbed_const(:transfer_nested_constants => true)
-  end
-  let(:validater) { instance_double(Catalog::ValidateSource) }
-  let(:valid_source) { true }
-
-  before do
-    allow(topological_inventory).to receive(:call).and_yield(api_instance)
-    allow(api_instance).to receive(:show_service_offering).with(service_offering_ref).and_return(topology_service_offering)
-    allow(api_instance).to receive(:show_service_offering_icon).with(topology_service_offering.service_offering_icon_id).and_return(service_offering_icon)
-    allow(Catalog::ValidateSource).to receive(:new).with(topology_service_offering.source_id).and_return(validater)
-    allow(validater).to receive(:process).and_return(validater)
-    allow(validater).to receive(:valid).and_return(valid_source)
+  let(:params) do
+    {
+      :name                 => "Frank",
+      :description          => "Franks Description",
+      :service_offering_ref => service_offering_ref
+    }
   end
 
-  context "user provided params" do
-    it "#process" do
-      ManageIQ::API::Common::Request.with_request(default_request) do
-        result = subject.process
-        expect(result.item.name).to eq(user_defined_name)
-        expect(result.item.description).to eq(user_defined_description)
+  around do |example|
+    ManageIQ::API::Common::Request.with_request(default_request) do
+      with_modified_env(:TOPOLOGICAL_INVENTORY_URL => "http://localhost", :SOURCES_URL => "http://localhost") do
+        example.call
       end
     end
+  end
+
+  describe "#process" do
+    let(:topology_service_offering) { fully_populated_service_offering }
+    let(:service_offering_icon) { fully_populated_service_offering_icon }
+    let(:validater) { instance_double(Catalog::ValidateSource) }
+
+    before do
+      stub_request(:get, "http://localhost/api/topological-inventory/v1.0/service_offerings/1")
+        .to_return(:status => 200, :body => topology_service_offering.to_json, :headers => default_headers)
+      stub_request(:get, "http://localhost/api/topological-inventory/v1.0/service_offering_icons/998")
+        .to_return(:status => 200, :body => service_offering_icon.to_json, :headers => default_headers)
+
+      allow(Catalog::ValidateSource).to receive(:new).with(topology_service_offering.source_id).and_return(validater)
```

Comment:
Technically we could mock this right? It would just be 2 hits to the sources API.
I did it here: https://github.com/lindgrenj6/catalog-api/blob/0f000f70c92aef09a1952f733c36de8910a64990/spec/services/catalog/validate_source_spec.rb#L15-L20